### DISCUSSION TITLE
Crossing the border of Languages in open source

### DISCUSSION THREAD

**Discussion by richakumar** [github/india#74] [Open Source]
Hi folks,

I will be chatting with @nickmalhotra, Head of Innovation at Tech Mahindra and creator of the Bharat Markup Language, and @knadh, CTO of Zerodha, and creator of Olam and Alar, open source regional language dictionaries. We will get a chance to hear from them about their efforts that are helping make open source more accessible to the non-English audience in India, something that is critical to advance the ecosystem.

We'll post some useful resources here and would love to help answer any questions.

**Comment by nickmalhotra** [comment]
Www.Bhaml.org

**Comment by knadh** [comment]
Thanks @richakumar. Here are the links to the resources mentioned in the talk.

**Dictionary websites**
- [Olam](https://olam.in) - English-Malayalam, Malayalam-Malayalam
- [Alar](https://alar.ink) - Kannada-English

**Code**
- [dictmaker](https://github.com/knadh/dictmaker) - Tool for building and publishing dictionaries in any language like the above ones.
- [ml2en](https://github.com/knadh/ml2en) - Algorithm for converting Malayalam to phonetic English (can be ported to other Indic languages).
- [mlphone](https://github.com/knadh/mlphone) - Algorithm for phonetically hashing Malayalam words (similar to Metaphone). Can be ported to other Indic languages.
- [knphone](https://github.com/knadh/knphone) - Algorithm for phonetically hashing Kannada words (similar to Metaphone). Can be ported to other Indic languages.
- [ditto](https://github.com/ajinasokan/ditto) - Flutter app framework for building mobile dictionary apps like the Olam, Alar apps.

**Open data**
- [Olam / Datuk Malayalam-Malayalam dictionary corpus](https://olam.in/open/datuk/) - 83,000 Malayalam words and 106,000 definitions.
- [Olam / English-Malayalam dictionary corpus](https://olam.in/open/enml/) - 61,000+ English words and 140,000 Malayalam definitions.
- [Alar / V. Krishna Kannada-English dictionary corpus](https://alar.ink/pages/about) - 150,000+ Kannada words and 240,000 English definitions.

The organisation I mentioned in the talk, the one that played a key role in enabling many Indic language computing technologies is [SMC (Swathanthra Malayalam Computing)](https://smc.org.in/projects]).

**Comment by nickmalhotra** [comment]
Thanks @richakumar and @knadh . It was a wonderful discussion this morning . Code repository is at https://github.com/nickmalhotra/BHARAT_CODE/pull/3 and the open source can be tried at http://BHaMl.org
